queries.sql

```diff
@@ -24,3 +24,52 @@ SELECT * from animals WHERE NOT name='Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * from animals WHERE weight_kg>=10.4 AND weight_kg<=17.3;
 SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals SET species='unspecified';
+SELECT name, species FROM animals;
+ROLLBACK;
+SELECT name, species FROM animals;
+
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+SELECT name, species FROM animals;
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+SELECT name, species FROM animals;
+COMMIT;
+SELECT name, species FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+SELECT name, species FROM animals;
+ROLLBACK;
+SELECT name, species FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth>'2022-01-01';
+SELECT name, date_of_birth FROM animals;
+SAVEPOINT saveAfterDelete;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO saveAfterDelete;
+SELECT * FROM animals;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg<0;
+SELECT name, weight_kg FROM animals;
+COMMIT;
+
+
+SELECT COUNT(*) as Number_of_Animals FROM animals;
+
+SELECT COUNT(*)  as Number_of_Animals_Never_Tried_To_Scape FROM animals WHERE escape_attempts=0;
+
+
+SELECT AVG(weight_kg) as Average_weight_of_animals FROM animals;
+
+
+SELECT COUNT(escape_attempts) AS Animal_escaped_the_most_count, neutered FROM animals GROUP BY neutered;
+
+
+SELECT MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight, species FROM animals GROUP BY species;
+
+
+SELECT AVG(escape_attempts) as Average_Escape_Attempt, species FROM animals WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-01-01' GROUP BY species;
```

Comment:
## OPTIONAL 
* I strongly suggest that you rewrite this query correctly 👇 

> What is the average number of escape attempts per animal type of those born between 1990 and 2000?
* Your current query is not accurate your data range should be `'1990/01/01' AND date_of_birth <= '2000/12/31'` this way you can properly cover animals born on the last day of year 2000

Reply:
Thanks 👌. Updated!